### PR TITLE
BaSP-TG-31: SuperAdmins Unit Tests

### PR DESCRIPTION
**Pull request by JulianFloresDev**
- GET all with filters
- EDIT
- CREATE

### CODE REVIEW

**Review by Licamus — DISMISSED** [review]

File: src/tests/superAdmins.test.js [hunk 1]
```diff
@@ -7,6 +7,103 @@ beforeAll(async () => {
...
+    test('if send a VALID path it should return a message like "Super Admin found"', async () => {
+      const response = await request(app).get('/api/superAdmins').send();
+      expect(response.body.message).toBe('Super Admin found');
+    });
+    test('if send an object with invalid properties on body it should return ALL Super Admins', async () => {
+      const response = await request(app).get('/api/superAdmins').send({ football: 'allways' });
+      expect(response.body.data.length).toBe(superAdminsSeed.length);
+    });
+    test('if send an object with invalid properties on body it should return 200 status code', async () => {
+      const response = await request(app).get('/api/superAdmins').send();
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/tests/superAdmins.test.js:64]: My bad, it's true!! it should be 400 status code!!

[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/tests/superAdmins.test.js:64]: Okay I understand! I think I will unify the two expected results in the same test.

File: src/tests/superAdmins.test.js [hunk 2]
```diff
@@ -89,3 +186,68 @@ describe('DELETE-BY-ID /api/superAdmins', () => {
...
+  describe('Success PUT tests', () => {
+    test('if send a VALID id as params & a VALID object as body it should return a valid request', async () => {
+      const response = await request(app).put(`/api/superAdmins/${id}`).send(superAdminValid);
+      expect(response.status).toBe(200);
+      expect(response.body.message).toBe(`SuperAdmin with the ID: ${id}, has been successfully edited!`);
+      expect(response.body.data).toMatchObject(superAdminValid);
+    });
+  });
+  describe('Wrong PUT tests', () => {
+    test('if send a VALID ID as params but an INVALID object as body ir shoul return a bad request', async () => {
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/tests/superAdmins.test.js:239]: Checked!! Tahnks

File: src/tests/superAdmins.test.js [hunk 3]
```diff
@@ -89,3 +186,68 @@ describe('DELETE-BY-ID /api/superAdmins', () => {
...
+      const response = await request(app).post('/api/superAdmins').send(superAdminInvalid.lastName);
+      expect(response.status).toBe(400);
+      expect(response.body).toHaveProperty('message');
+    });
+    test('if send an invalid proeprty value it should return status code 400', async () => {
+      const response = await request(app).post('/api/superAdmins').send(superAdminInvalid.email);
+      expect(response.status).toBe(400);
+      expect(response.body).toHaveProperty('message');
+    });
+    test('if send an invalid proeprty value it should return status code 400', async () => {
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/tests/superAdmins.test.js:220]: I tried to be generic in order to repeat the test several times below with different properties

File: src/tests/superAdmins.test.js [hunk 4]
```diff
@@ -89,3 +186,68 @@ describe('DELETE-BY-ID /api/superAdmins', () => {
...
+      expect(response.body.message).toBe(`SuperAdmin with the ID: ${id}, has been successfully edited!`);
+      expect(response.body.data).toMatchObject(superAdminValid);
+    });
+  });
+  describe('Wrong PUT tests', () => {
+    test('if send a VALID ID as params but an INVALID object as body ir shoul return a bad request', async () => {
+      const response = await request(app).put(`/api/superAdmins/${id}`).send(superAdminInvalid.name);
+      expect(response.status).toBe(400);
+    });
+    test('if send a VALID ID as params but an INVALID object as body ir shoul return a bad request', async () => {
```
[the remark this answers is not in the document]
Comment (JulianFloresDev) [reply src/tests/superAdmins.test.js:243]: Done

**Review by ignacioaristo — APPROVED** [review]

**Pull request merged**